subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt

```diff
@@ -321,6 +321,21 @@ abstract class InHouseInstrumentationTestRunner :
         arguments.putString("newRunListenerMode", "true")
     }
 
+    override fun finish(resultCode: Int, results: Bundle?) {
+        try {
+            super.finish(resultCode, results)
+        } catch (e: IllegalStateException) {
+            // we made a research.
+            // It showed that IllegalStateException("UiAutomation not connected") occurs unrelated to our code.
+            // We use uiAutomation only for VideoCapture but without capturing this exception occurs with same frequency
+            if (e.message?.contains("UiAutomation not connected") != true) {
```

Comment:
Let's check also the exception class to be sure

Reply:
I think `catch` did it for us

Reply:
@eugene-krivobokov Is this enough?

Reply:
sorry, missed that we are already in catch block